PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutFormFragment.java

```diff
@@ -97,19 +110,46 @@ public void onClick(View view){
             /**
              * Loads next fragment onto the current stack.
              */
-            FragmentTransaction transaction =
-                    ((Activity)mContext).getFragmentManager().beginTransaction();
-                    SuccessFragment successFragment = new SuccessFragment();
-                    successFragment.setType(SuccessFragment.SuccessType.CHECKOUT_SUCCESS);
-                    transaction.replace(R.id.checkout_activity_container, successFragment);
-            transaction.commit();
+
+            Bundle args = new Bundle();
+            args.putCharSequence("scan_result", mScanResult);
+            args.putBoolean("manual_input", mManualInput);
+            if(mComment == null ){
+                args.putString("comments",  " ");
+            }else{
+                args.putString("comments",  mComment.getText().toString());
+            }
+            args.putInt("experience", mExperience);
+
+
+
+            CheckoutConfirmationFragment confFrag = new CheckoutConfirmationFragment();
+            confFrag.setArguments(args);
+            displayNextFragment(confFrag, CheckoutConfirmationFragment.TAG);
         }
     }
 
 
+
+    /**
+     * Brings up another fragment when this fragment
+     * is complete
+     * @param nextFrag Fragment to display next
+     * @param fragName String fragment names
+     */
+
+    protected void displayNextFragment(Fragment nextFrag, String fragName) {
+        FragmentTransaction transaction =
+                (getActivity()).getFragmentManager().beginTransaction();
```

Comment:
since you call getActivity() 4 times in this class, you can set it as a class variable in an oncreate